subsys/bluetooth/services/ras/ras_internal.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef BT_RAS_INTERNAL_H_
+#define BT_RAS_INTERNAL_H_
+
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <bluetooth/services/ras.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define RASCP_CMD_OPCODE_LEN     1
+#define RASCP_CMD_OPCODE_OFFSET  0
+#define RASCP_CMD_PARAMS_OFFSET  RASCP_CMD_OPCODE_LEN
+#define RASCP_CMD_PARAMS_MAX_LEN 4
+#define RASCP_WRITE_MAX_LEN      (RASCP_CMD_OPCODE_LEN + RASCP_CMD_PARAMS_MAX_LEN)
+
+/** @brief RAS Control Point opcodes as defined in RAS Specification, Table 3.10. */
+enum rascp_opcode {
+	RASCP_OPCODE_GET_RD                    = 0x00,
+	RASCP_OPCODE_ACK_RD                    = 0x01,
+	RASCP_OPCODE_RETRIEVE_LOST_RD_SEGMENTS = 0x02,
+	RASCP_OPCODE_ABORT_OP                  = 0x03,
+	RASCP_OPCODE_SET_FILTER                = 0x04,
+};
+
+/** @brief RAS Control Point Response opcodes as defined in RAS Specification, Table 3.11. */
+enum rascp_rsp_opcode {
+	RASCP_RSP_OPCODE_COMPLETE_RD_RSP          = 0x00,
+	RASCP_RSP_OPCODE_COMPLETE_LOST_RD_SEG_RSP = 0x01,
+	RASCP_RSP_OPCODE_RSP_CODE                 = 0x02,
+};
+
+#define RASCP_RSP_OPCODE_COMPLETE_RD_RSP_LEN          2
+#define RASCP_RSP_OPCODE_COMPLETE_LOST_RD_SEG_RSP_LEN 4
+#define RASCP_RSP_OPCODE_RSP_CODE_LEN                 1
+
+/** @brief RAS Control Point Response Code Values as defined in RAS Specification, Table 3.12. */
+enum rascp_rsp_code {
+	RASCP_RESPONSE_RESERVED                = 0x00,
```

Comment:
we could consider removing this one (since there's also no RASCP_RESPONSE_RESERVED for the range 0x09-0xFF)

Reply:
I don't have a strong preference here, other nrf profiles generally follow the convention of keeping it in the enum but I can remove.

Reply:
seems reasonable then. It's not so important